src/presenter/film-presenter.js

```diff
@@ -1,128 +1,122 @@
-    if (this.#films.length > ALL_FILM_COUNT_PER_STEP) {
-      render(this.#loadMoreButtonComponent, this.#allFilmListComponent.element);
-      this.#loadMoreButtonComponent.setClickHandler(this.#handleLoadMoreButtonClick);
-    }
-    this.#films.sort((a, b) => b.filmInfo.totalRating - a.filmInfo.totalRating).slice(0, Math.min(this.#films.length, TOP_RATED_FILM_COUNT_PER_STEP)).forEach((film) => this.#renderFilm(film, this.#topRatedFilmListContainerComponent.element));
-    this.#films.sort((a, b) => b.comments.length - a.comments.length).slice(0, Math.min(this.#films.length, MOST_COMMENTED_FILM_COUNT_PER_STEP)).forEach((film) => this.#renderFilm(film, this.#mostCommentedFilmListContainerComponent.element));
-  }
+    this.#filmCardComponent.setClickHandler(this.#OpenfilmDetails);
+    this.#filmCardComponent.setWatchListClickHandler(this.#handleWatchListClick);
+    this.#filmCardComponent.setWatchedClickHandler(this.#handleWatchedClick);
+    this.#filmCardComponent.setFavoriteClickHandler(this.#handleFavoriteClick);
+    this.#filmDetailsComponent.setCloseButtonClickHandler(this.#ClosefilmDetails);
+    this.#filmDetailsComponent.setWatchListClickHandler(this.#handleWatchListClick);
+    this.#filmDetailsComponent.setWatchedClickHandler(this.#handleWatchedClick);
+    this.#filmDetailsComponent.setFavoriteClickHandler(this.#handleFavoriteClick);
 
-  #renderFilm (film, FilmCountainerElement) {
-    const filmCardComponent = new FilmCardView(film);
-    render(filmCardComponent, FilmCountainerElement);
+    if (prevFilmCardComponent === null) {
+      return render(this.#filmCardComponent, this.#filmCountainerElement);
+    }
 
-    const onCardClick = () => {
-      this.#renderFilmDetails(film);
-    };
+    replace(this.#filmCardComponent, prevFilmCardComponent);
+    if (this.#mode === Mode.OPEND) {
+      replace(this.#filmDetailsComponent, prevFilmDetailsComponent);
+    }
 
-    filmCardComponent.setClickHandler(onCardClick);
+    remove(prevFilmCardComponent);
+    remove(prevFilmDetailsComponent);
   }
 
-  #renderFilmDetails (film) {
-    this.#filmDetailsComponent = new FilmDetailsView(film);
-    render(this.#filmDetailsComponent, document.body);
-    this.#filmModel.getCommentsByFilm(film.id).forEach((comment) => render(new FilmDetailsCommentView(comment), this.#filmDetailsComponent.element.querySelector('.film-details__comments-list')));
+  destroy = () => {
+    remove(this.#filmCardComponent);
+    remove(this.#filmDetailsComponent);
+  };
 
-    const escKeydownHandler = (evt) => {
-      if (evt.key === 'Escape' || evt.key === 'Esc') {
-        evt.preventDefault();
-        document.body.classList.remove('hide-overflow');
-        this.#handleClosefilmDetails();
-        document.removeEventListener('keydown', escKeydownHandler);
-      }
-    };
+  #OpenfilmDetails = () => {
```

Comment:
за что такая честь ?

```suggestion
  #оpenfilmDetails = () => {
```